dist/contract.js

```diff
@@ -122,39 +126,43 @@ var Contract = function () {
 
       var hex = void 0;
       _lodash2.default.each(methodObj.inputs, function (item, index) {
-        switch (item.type) {
-          case 'address':
-            {
-              hex = _encoder2.default.addressToHex(args[index]);
+        var type = item.type;
+
+        if (type === 'address') {
+          hex = _encoder2.default.addressToHex(args[index]);
+          dataHex = dataHex.concat(hex);
+        } else if (type === 'bool') {
+          hex = _encoder2.default.boolToHex(args[index]);
+          dataHex = dataHex.concat(hex);
+        } else if (type.startsWith('uint')) {
```

Comment:
if we use regex (or type set as I suggestion) to match `bytes` type, probably I should match similarily on `int` & `uint` too

Reply:
done